core/src/state/stateful.rs

```diff
@@ -56,7 +60,11 @@ impl<W: 'static> StateReader for Stateful<W> {
   fn read(&self) -> ReadRef<W> { ReadRef::new(self.data.borrow()) }
 
   #[inline]
-  fn clone_reader(&self) -> Self::Reader { Reader(self.clone()) }
+  fn clone_reader(&self) -> Self::Reader {
+    let this = self.clone();
+    this.dec_writer();
```

Comment:
You can't call `dec_writer`, otherwise the inner `Stateful` may destroy before the reader.
```rust
let a = Stateful::new(0);
let b  = a.clone_reader();
drop(a);

// `b`  is not valid now.


```

Reply:
![image](https://github.com/RibirX/Ribir/assets/6829874/9e3b78c7-d663-4d81-9b31-8dfe8a09ad38)
the drop for reader has already call dec_writer . it's bug, if not call at clone_reader now

Reply:
You are right. I forget the writer has a different reference count, my example will work.